backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java

```diff
@@ -294,12 +297,34 @@ private void linkInvoiceToPaymentIfNeeded(final I_C_Invoice invoice)
 		}
 	}
 
+	@VisibleForTesting
+	static boolean canAllocateOrderPaymentToInvoice(final I_C_Order order)
+	{
+		if (order == null)
+		{
+			return false;
+		}
+		if (order.getC_Payment_ID() <= 0)
+		{
+			return false;
+		}
+
+		final boolean isPrepayOrder = Services.get(IDocTypeBL.class).isPrepay(DocTypeId.ofRepoId(order.getC_DocType_ID()));
+		if (isPrepayOrder)
+		{
+			return true;
+		}
+
+		final ExternalId paymentExternalId = Services.get(IPaymentDAO.class).getExternalId(PaymentId.ofRepoId(order.getC_Payment_ID()));
```

Comment:
❗ why do the external IDs have to match? Mark doesn't know, maybe you discussed this with @metasnw ?

To put it differently:
* you have a sales order that reverences a payment.
* you invoice this sales order
* => why not allocate the sales order's payment to the invoice?

If you need to keep it like this, please add it to the documentation of `C_Order.C_Payment_ID`, 'C_Order.ExternalId` and `C_Payment.ExternalOrderId`, because i think it's intuitive to "forward" the referenced payment from the order..but if it's only forwarded if the external-ids match, then this needs to be be documented.

Reply:
confirmed with torby: we allow any payment linked to the Sales Order, not just those with matching external ID.